### PR TITLE
fix zstd on windows

### PR DESCRIPTION
**Pull request by cha5on**
Sandboxing on Windows is more limited than on other platforms, so two jni_md.h headers were ending up in the include path.  This resulted in the wrong jni_md.h header being used on Windows, which meant that the JNIEXPORT macro was not being set to actually export symbols appropriately on Windows.  This caused any attempts to use zstd on windows, whether for http_archive, gRPC cache compression, etc. to result in an UnsatisfiedLinkError being thrown.

Change to use a similar method as the main bazel jni stuff, which is to copy the jni.h and jni_md.h headers over.  This prevents accidentally picking up a header from the wrong architecture.

Relates to #16041

### CODE REVIEW

File: third_party/zstd-jni/zstd-jni.BUILD [hunk 1]
```diff
@@ -1,18 +1,32 @@
+genrule(
+    name = "copy_link_jni_md_header",
+    srcs = select({
+        "@io_bazel//src/conditions:darwin": ["@bazel_tools//tools/jdk:jni_md_header-darwin"],
```
Comment (meteorcloudy) [inline third_party/zstd-jni/zstd-jni.BUILD:4]: `@bazel_tools//src/conditions:*` is better.

**Review by meteorcloudy — CHANGES_REQUESTED** [review]

File: third_party/zstd-jni/zstd-jni.BUILD [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
     name = "libzstd-jni.so",
     srcs = glob([
         "src/main/native/**/*.c",
         "src/main/native/**/*.h",
-    ]) + select({
-        "@io_bazel//src/conditions:windows": [
-            "src/windows/include/jni_md.h",
-            "jni/jni.h",
+    ]) + [
+            "jni_md.h",
```
Comment (coeuvre) [inline third_party/zstd-jni/zstd-jni.BUILD:27]: prepend `:`? e.g. `:jni_md.h`

**Comment by meteorcloudy** [comment]
We should try to enable the tests on Windows as well: https://cs.opensource.google/bazel/bazel/+/master:.bazelci/presubmit.yml;l=244-245

**Review by coeuvre — CHANGES_REQUESTED** [review]

File: third_party/zstd-jni/zstd-jni.BUILD (hunk 1 above)
Comment (gregoryT5) [reply third_party/zstd-jni/zstd-jni.BUILD]: updated throughout

File: third_party/zstd-jni/zstd-jni.BUILD (hunk 2 above)
Comment (gregoryT5) [reply third_party/zstd-jni/zstd-jni.BUILD]: done

File: third_party/zstd-jni/zstd-jni.BUILD (hunk 1 above)
Comment (cha5on) [reply third_party/zstd-jni/zstd-jni.BUILD]: updated throughout

File: third_party/zstd-jni/zstd-jni.BUILD (hunk 2 above)
Comment (cha5on) [reply third_party/zstd-jni/zstd-jni.BUILD]: done

**Comment by meteorcloudy** [comment]
Can you also update https://cs.opensource.google/bazel/bazel/+/master:.bazelci/postsubmit.yml;l=252 to the same as the presubmit.yml file?

**Review by coeuvre — APPROVED** [review]
LGTM. Thanks!

**Comment by cha5on** [comment]
> Can you also update https://cs.opensource.google/bazel/bazel/+/master:.bazelci/postsubmit.yml;l=252 to the same as the presubmit.yml file?

done!

**Review by meteorcloudy — APPROVED** [review]
Thank you so much for the fix!

**Pull request closed without merging**